ephios/plugins/questionnaires/models.py

```diff
@@ -0,0 +1,214 @@
+                    initial = ", ".join(initial)
+            case self.Type.SINGLE:
+                if isinstance(initial, list):
+                    # Reset answer if the question was a multiple choice question before
+                    if len(initial) == 1:
+                        initial = initial[0]
+                    else:
+                        initial = None
+
+                if initial not in self.choices:
+                    # Reset answer if the option does not exist any more
+                    initial = None
+            case self.Type.MULTIPLE:
+                if isinstance(initial, str):
+                    # Convert single answer to multi-answer
+                    initial = [initial]
+
+                if isinstance(initial, list):
+                    # Remove invalid answers if answers are set
+                    initial = [answer for answer in initial if answer in self.choices]
+
+        return initial
+
+    def get_signup_form_field(
+        self, participant: AbstractParticipant, participation: AbstractParticipation
+    ):
+        initial = self._get_initial_answer(participant, participation)
+
+        field_name = self.get_form_slug()
+        required = self.required if initial is None else False
+
+        field_classes = self._get_field_classes()
+        field_kwargs = self._get_field_kwargs()
+
+        field = {
+            "form_class": field_classes[0],
+            "form_kwargs": {
+                "label": self.question_text,
+                "help_text": self.description,
+                "initial": initial,
+                "required": required,
+                **field_kwargs[0],
+            },
+            "serializer_class": field_classes[1],
+            "serializer_kwargs": {
+                "required": required,
+                **field_kwargs[1],
+            },
+        }
+
+        return field_name, field
+
+    def get_saved_answer_form_field(self):
+        field_classes = self._get_field_classes()
+        field_kwargs = self._get_field_kwargs()
+
+        return field_classes[0](**field_kwargs[0])
+
+    def get_form_slug(self):
+        return "questionnaires_" + slugify(f"{self.pk} {self.name}")
+
+    @staticmethod
+    def get_pk_from_slug(slug: str):
+        """
+        Returns the questions' pk from a form field slug  generated by `Question.get_form_slug`.
+        These slugs have the format `questionnaires_123-demo-question`.
+
+        For this slug, this method would return `123`.
+        """
+        return int(slug.split("_", maxsplit=1)[1].split("-")[0])
+
+
+class Questionnaire(models.Model):
+    shift = models.OneToOneField(Shift, on_delete=models.CASCADE)
+    questions = models.ManyToManyField(Question, blank=True)
+
+    class Meta:
+        verbose_name = _("Questionnaire")
+        verbose_name_plural = _("Questionnaires")
+
+    def __str__(self):
+        return f"{", ".join(self.questions.values_list("name", flat=True))} @ {self.shift}"
```

Comment:
broken in CI: beware that before python3.12, you can't mix quotation styles in f-strings and their substitution fields:


```suggestion
        return f'{", ".join(self.questions.values_list("name", flat=True))} @ {self.shift}'
```